### PR TITLE
chore: write-gcm option and enable/disable agent

### LINKED ISSUE #15: Support the --write-gcm installation option.
I think this is just a matter of adding `DETECT_GCM="yes"` to `stackdriver.sysconfig` and dropping the API key, but for EC2 we will also need to support uploading of the credentials JSON file to the host.

### PR DESCRIPTION
**Pull request by Jamal-B**
As #23, this also partially fixes #15, but without removing api key support because some users still need it. This branch also add the possibility to disable/enable the stackdriver agent.

### CODE REVIEW

**Comment by Jamal-B** [comment]
I signed it!

File: tasks/configure-agent.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 ---
-- name: Configure the agent with the Stackdriver API key.
+- name: Configure the agent.
   template: src=stackdriver.sysconfig dest={{ stackdriver_sysconfig }} backup=yes
   notify:
```
Comment (dhrupadb) [inline tasks/configure-agent.yml]: `- restart extractor` throws an error as it is not found. 
Change it to match up with handlers/main.yml: Line 18 

`name: restart extractor if not using gcm` 

or revert it to the original 

`name: restart extractor`

Comment (Jamal-B) [inline tasks/configure-agent.yml]: Indeed, you are right. I finally changed the name in configure-agent.yml to be more explicit for gcm users.

**Comment by Jamal-B** [comment]
Sorry for this dirty git historic (multiple fixup + rebase autosquash + tagged release in the fork!). If needed I can start from zero a new branch/pull request.

**Comment by dhrupadb** [comment]
The changes now look good. I will say that the ability to enable and disable the agent is over configuring the role. It is easy enough from a user perspective to disable the agent in their main playbook rather than find the particular flag in the role. Could you perhaps then revert commit :8bbe3b5b33451ae8d052e5b8ed14b8d22e58d99d ?

**Comment by igorpeshansky** [comment]
@Jamal-B Is anything happening with this PR? We're ok with the dirty history -- we'll squash on merging anyway. Can you please address the remaining comments so we can merge this?

File: README.md [hunk 2]
```diff
@@ -14,17 +14,24 @@ git clone this-git-repo roles/stackdriver
...
+```
+stackdriver_use_gcm: true
+```
+
+Legacy authentication method: provide stackdriver API key in your playbook:
+```
+stackdriver_api_key: "API_KEY"
+```
 
+You can choose to don't start stackdriver agent (and extractor if not using gcm) after installation or to stop it if it is running by setting stackdriver_enabled variable to false (the default value is `true`):
```
Comment (igorpeshansky) [inline README.md]: s/`to don't`/`to not`/. s/`stackdriver`/`Stackdriver`/.

File: defaults/main.yml [hunk 3]
```diff
@@ -16,6 +16,13 @@
 # Please refer to ../README.md for more information about configuring the
 # various plugins.
 
+### stackdriver agent options
```
Comment (igorpeshansky) [inline defaults/main.yml]: s/`stackdriver`/`Stackdriver`/

File: defaults/main.yml [hunk 4]
```diff
@@ -16,6 +16,13 @@
...
+
+# define if stackdriver agent must be running or not (more info in the readme)
```
Comment (igorpeshansky) [inline defaults/main.yml]: s/`stackdriver`/`Stackdriver`/

File: tasks/configure.yml [hunk 5]
```diff
@@ -12,15 +12,22 @@
...
-  service: name={{ stackdriver_collectd_service }} state=started enabled=yes
+- name: "Ensure collectd is {% if not stackdriver_enabled %}not {% endif %}running."
+  service:
+    name: "{{ stackdriver_collectd_service }}"
+    state: "{% if stackdriver_enabled %}started{% else %}stopped{% endif %}"
+    enabled: "{% if stackdriver_enabled %}yes{% else %}no{% endif %}"
 
-- name: Ensure extractor is running.
-  service: name={{ stackdriver_extractor_service }} state=started enabled=yes
+- name: "Ensure extractor is {% if not (stackdriver_enabled and not stackdriver_use_gcm) %}not {% endif %}running."
```
Comment (igorpeshansky) [inline tasks/configure.yml]: `{% if stackdriver_use_gcm or not stackdriver_enabled %}` (simpler and consistent with below).

File: tasks/configure.yml [hunk 6]
```diff
@@ -12,15 +12,22 @@
...
+  service:
+    name: "{{ stackdriver_extractor_service }}"
+    state: "{% if stackdriver_use_gcm or not stackdriver_enabled %}stopped{% else %}started{% endif %}"
+    enabled: "{% if stackdriver_use_gcm or not stackdriver_enabled %}no{% else %}yes{% endif %}"
 
-- name: Determine the Stackdriver host ID for reporting.
+- name: Determine the Stackdriver host ID for reporting if not using gcm
```
Comment (igorpeshansky) [inline tasks/configure.yml]: You've lost the trailing ".".

File: tasks/main.yml [hunk 7]
```diff
@@ -12,9 +12,9 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 ---
-- name: Check that the Stackdriver API key has been specified.
+- name: Check that the Stackdriver API key has been specified if not using gcm.
   fail: msg="Please provide your Stackdriver API key in the variable stackdriver_api_key."
-  when: stackdriver_api_key is not defined or not stackdriver_api_key
+  when: (stackdriver_use_gcm is not defined and stackdriver_use_gcm) and (stackdriver_api_key is not defined or not stackdriver_api_key)
```
Comment (igorpeshansky) [inline tasks/main.yml]: Is it just me, or is `and stackdriver_use_gcm` superfluous here?

File: templates/stackdriver.sysconfig [hunk 8]
```diff
@@ -4,5 +4,7 @@ AUTOGENERATE_COLLECTD_CONFIG="yes"
 # url to a proxy for outbound https
 PROXY_URL=""
 
-# your stackdriver api key
-STACKDRIVER_API_KEY="{{ stackdriver_api_key }}"
+# your stackdriver api key if not using gcm
```
Comment (igorpeshansky) [inline templates/stackdriver.sysconfig]: How about:
```jinja
{% if stackdriver_api_key is defined %}
# your stackdriver api key
STACKDRIVER_API_KEY="{{ stackdriver_api_key }}"
{% endif %}

{% if stackdriver_use_gcm %}
DETECT_GCM="yes"
{% endif %}
```
?

**Pull request closed without merging**